carbonmark/components/pages/Project/Retire/Pool/RetireInputs.tsx

```diff
@@ -9,35 +10,93 @@ import { TextareaField } from "components/shared/Form/TextareaField";
 import { Text } from "components/Text";
 import { utils } from "ethers";
 import { formatToPrice, formatToTonnes } from "lib/formatNumbers";
-import { carbonmarkPaymentMethodMap } from "lib/getPaymentMethods";
+import { carbonmarkRetirePaymentMethodMap } from "lib/getPaymentMethods";
 import {
   CarbonmarkPaymentMethod,
   Price as PriceType,
 } from "lib/types/carbonmark";
 import Image from "next/legacy/image";
 import { useRouter } from "next/router";
-import { FC } from "react";
-import { SubmitHandler, useFormContext } from "react-hook-form";
+import { FC, useEffect } from "react";
+import { SubmitHandler, useFormContext, useWatch } from "react-hook-form";
 import * as styles from "./styles";
 import { FormValues } from "./types";
 
 type Props = {
   onSubmit: (values: FormValues) => void;
   price: PriceType;
   values: null | FormValues;
-  balance: string | null;
+  userBalance: string | null;
+  fiatBalance: string | null;
+  address?: string;
 };
 
+const validations = (
+  userBalance: string | null,
+  fiatBalance: string | null
+) => ({
+  usdc: {
+    quantity: {
+      min: {
+        value: 0.001,
+        message: t`The minimum amount to retire is 0.001 Tonnes`,
+      },
+    },
+    totalPrice: {
+      max: {
+        value: Number(userBalance || "0"),
+        message: t`You exceeded your available amount of tokens`,
+      },
+    },
+  },
+  fiat: {
+    quantity: {
+      min: {
+        value: 1,
+        message: t`The minimum amount to retire is 1 Tonnes`,
+      },
+    },
+    totalPrice: {
+      max: {
+        value: Number(fiatBalance || "2000"),
+        message: t`At this time, Carbonmark cannot process credit card payments exceeding ${formatToPrice(
+          fiatBalance || 0
```

Comment:
I'm having second thoughts about this. I think we should go back to only referencing the MAX_USDC value. We can expose the max value through the new endpoint you created, so Offsetra can change it without having to update the klimadao code.

We can remove the balance checks completely.
- Offsetra and team should always ensure there is at least $2k or more in liquid USDC.
- Clients should not have to care about this balance at all, they just need to know the max allowed. In very rare cases where it gets bought out, checkouts will simply fail until Offsetra can replenish it. They have alerts set up already.
- The rule should actually become `$2k per user per day` or `${MAX_USDC} per user per day` and we should update the error message to say that instead. And some day in the future we could even enforce this by looking up payment records in firebase.

Reply:
I replaced the balance endpoint on the Offsetra API to a new endpoint "info".
This endpoint gives back the MAX_USDC value.

When on Production the value of MAX_USDC is hardcoded to 2000.
When on Staging the actual server wallet balance is returned. 
Otherwise the UI does not make sense (Api throws very quickly, but everywhere it says $2.000 ? Rather show here the real max balance of the connected Server Wallet.)